airflow-core/src/airflow/api_fastapi/core_api/routes/public/dag_run.py

```diff
@@ -156,7 +157,7 @@ def patch_dag_run(
             f"The DagRun with dag_id: `{dag_id}` and run_id: `{dag_run_id}` was not found",
         )
 
-    dag: DAG = request.app.state.dag_bag.get_dag(dag_id)
+    dag: DAG = get_dag_from_dag_bag(request.app.state.dag_bag, dag_id)
```

Comment:
typing is wrong `DAG | None` (This is found in multiple places and need to be updated everywhere)